metaflow/mflog/mflog.py

```diff
@@ -113,11 +113,9 @@ def refine(line, prefix=None, suffix=None):
     prefix = to_bytes(prefix) if prefix else b''
     suffix = to_bytes(suffix) if suffix else b''
     parts = line.split(b']', 1)
-    if len(parts) == 2:
-        header, body = parts
-        return b''.join((header, b']', prefix, body, suffix))
-    else:
-        return line
+    assert len(parts) == 2
```

Comment:
I am not a big fan of asserts. We can probably leave this segment of the code as-is for this PR.

Reply:
That's the bit we talked about offline, it seems like this is always called with mflog-formatted line so that `else` branch doesn't get called anyway. I can remove the assert itself given that it will fail on the next line anyway tho

Reply:
addressed in #504